src/nspawn/nspawn.c

```diff
@@ -1104,6 +1105,8 @@ static int parse_argv(int argc, char *argv[]) {
         if (e)
                 arg_container_service_name = e;
 
+        arg_use_cgns = getenv_bool("SYSTEMD_NSPAWN_USE_CGNS");
```

Comment:
Shouldn't it be the reverse?

If we get -ENXIO, it means the variable is not set. In this case we should default to use cgns if it is supported. But the code below disables cgns when the variable is not set.

Also, shouldn't the check for error be `< 0` instead of `== -ENXIO`? Checks for errors are usually `< 0` in systemd.

```
r = getenv_bool("SYSTEMD_NSPAWN_USE_CGNS");
if (r < 0)
    arg_use_cgns = cg_ns_supported();
else
    arg_use_cgns = r;
```